Monika After Story/game/special-effects.rpy

```diff
@@ -1,5 +1,669 @@
+
+        @property
+        def z(self):
+            return self._z
+
+        @z.setter
+        def z(self, value):
+            self._z = value
+            self.update_offsets()
+
+        @property
+        def zoom(self):
+            return self._zoom
+
+        @zoom.setter
+        def zoom(self, value):
+            value = min(max(value, self.min_zoom), self.max_zoom)
+            self._zoom = value
+            self._transform.zoom = value
+            self.update_offsets()
+
+        def update_mouse_pos(self):
+            """
+            Updates mouse pos
+            """
+            self.mouse_x, self.mouse_y = renpy.get_mouse_pos()
+
+        def reset_mouse_pos(self):
+            """
+            Resets mouse pos
+            """
+            self.mouse_x = int(renpy.config.screen_width / 2.0)
+            self.mouse_y = int(renpy.config.screen_height / 2.0)
+
+        def toggle_events(self, value):
+            """
+            Toggles events (and hence the parallax effect)
+            """
+            self._enable_events = value
+
+        def __repr__(self):
+            """
+            Representation of this object
+            """
+            return "<{0}: (img: {1}, x: {2}, y: {3}, z: {4}, decals: {5})>".format(type(self).__name__, self._container.base, self._x, self._y, self._z, self._container.children)
+
+        def update_offsets(self):
+            """
+            Updates the offsets of this parallax sprite
+            NOTE: I have no idea how I made this meth work, don't change it
+            """
+            screen_width = renpy.config.screen_width
+            screen_height = renpy.config.screen_height
+
+            zoom_factor = abs(self._zoom - ParallaxSprite.NORMAL_ZOOM)
+
+            zoom_correction_x = screen_width * zoom_factor / 2.0
+            zoom_correction_y = screen_height * zoom_factor / 2.0
+
+            # We use screen_width and screen_height for our parallax
+            available_x_shift = screen_width / float(self._z)
+            available_y_shift = screen_height / float(self._z)
+
+            half_screen_width = screen_width / 2.0
+            half_screen_height = screen_height / 2.0
+
+            mouse_x_factor = self.mouse_x / half_screen_width
+            mouse_y_factor = self.mouse_y / half_screen_height
+
+            container_offset_x = abs(self._container.offsets[0])
+            container_offset_y = abs(self._container.offsets[1])
+
+            # Our offsets consist of 4 parts:
+            # - base coords give offsets to x and y (depend on zoom level)
+            # - shift from the parallax effect (depends on mouse pos)
+            # - correction to the zoom effect (depends on mouse pos)
+            # - offsets from the container (depend on zoom level)
+            self._transform.xoffset = (
+                self._x*(1.0 + zoom_factor)
+                + available_x_shift*(1.0 - mouse_x_factor)
+                - zoom_correction_x*mouse_x_factor
```

Comment:
```suggestion
                - zoom_factor * self.mouse_x
```

`zoom_correction` is `(screen dimension/ 2) * zoom factor`

`mouse_factor` is `(2 / screen dimension) * mouse pos`

Since this is multiplying these together, the `screen dimension` and `2` cancel out, and the resulting calculation is just `zoom factor * mouse pos`. This also makes more sense conceptually, as the offset is adjusted by the position of the mouse but is dependent on how much zoom currently exists.